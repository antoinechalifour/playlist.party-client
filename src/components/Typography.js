import styled, { css } from 'styled-components'

const typographyThemes = {
  default: css`
    color: ${({ theme }) => theme.colors.typography};
  `,
  light: css`
    color: ${({ theme }) => theme.colors.typographyLight};
  `
}

const variantThemes = {
  default: css``,
  display1: css`
    font-size: 2.5rem;
  `
}

const Typography = styled.p`
  line-height: 1.6;
  
  ${({ type }) => typographyThemes[type]}
  ${({ variant }) => variantThemes[variant]}
`

Typography.defaultProps = {
  type: 'default',
  variant: 'default'
}

export default Typography
