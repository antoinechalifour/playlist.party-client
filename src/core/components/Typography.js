import styled, { css } from 'styled-components'

const typographyThemes = {
  default: css``,
  secondary: css`
    opacity: .65;
  `
}

const variantThemes = {
  default: css``,
  display1: css`
    font-size: 2.5rem;
  `,
  display2: css`
    font-size: 2rem;
  `,
  h2: css`
    font-size: 1.3rem;
    font-weight: bold;
  `
}

const Typography = styled.p`
  line-height: 1.6;
  color: ${({ theme, reverse }) => (reverse ? '#fff' : theme.colors.typography)};

  ${({ type }) => typographyThemes[type]}
  ${({ variant }) => variantThemes[variant]}
`

Typography.defaultProps = {
  reverse: false,
  type: 'default',
  variant: 'default'
}

export default Typography
