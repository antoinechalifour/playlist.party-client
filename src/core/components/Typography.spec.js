import React from 'react'
import renderer from 'react-test-renderer'
import Typography from './Typography'
import 'jest-styled-components'

const theme = {
  colors: {
    dark: '#151515',
    danger: '#a63d40',
    primary: '#1db954',
    confirm: '#90a959',
    accent: '#6494aa',
    typography: '#373d3f',
    typographyLight: '#bcbcbc'
  }
}

describe('Typography', () => {
  it('Should render default typography', () => {
    const tree = renderer
      .create(<Typography theme={theme}>Hello world :D</Typography>)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('Should render secondary typography', () => {
    const tree = renderer
      .create(
        <Typography type='secondary' theme={theme}>Hello world :D</Typography>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  const reverse = [true, false]
  const types = ['default', 'secondary']
  const variants = ['default', 'display1', 'display2', 'h2']

  reverse.forEach(isReverse =>
    types.forEach(type =>
      variants.forEach(variant =>
        it(`Should render ${isReverse ? 'reverse' : 'normal'} ${type} ${variant} typography`, () => {
          const tree = renderer
            .create(
              <Typography
                theme={theme}
                reverse={isReverse}
                type={type}
                variant={variant}
              >
                Hello world ;)
              </Typography>
            )
            .toJSON()

          expect(tree).toMatchSnapshot()
        })
      )
    )
  )
})
