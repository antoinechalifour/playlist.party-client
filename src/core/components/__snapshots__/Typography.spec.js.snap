// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography Should render default typography 1`] = `
.c0 {
  line-height: 1.6;
  color: #373d3f;
}

<p
  className="c0"
  type="default"
>
  Hello world :D
</p>
`;

exports[`Typography Should render normal default default typography 1`] = `
.c0 {
  line-height: 1.6;
  color: #373d3f;
}

<p
  className="c0"
  type="default"
>
  Hello world ;)
</p>
`;

exports[`Typography Should render normal default display1 typography 1`] = `
.c0 {
  line-height: 1.6;
  color: #373d3f;
  font-size: 2.5rem;
}

<p
  className="c0"
  type="default"
>
  Hello world ;)
</p>
`;

exports[`Typography Should render normal default display2 typography 1`] = `
.c0 {
  line-height: 1.6;
  color: #373d3f;
  font-size: 2rem;
}

<p
  className="c0"
  type="default"
>
  Hello world ;)
</p>
`;

exports[`Typography Should render normal default h2 typography 1`] = `
.c0 {
  line-height: 1.6;
  color: #373d3f;
  font-size: 1.3rem;
  font-weight: bold;
}

<p
  className="c0"
  type="default"
>
  Hello world ;)
</p>
`;

exports[`Typography Should render normal secondary default typography 1`] = `
.c0 {
  line-height: 1.6;
  color: #373d3f;
  opacity: .65;
}

<p
  className="c0"
  type="secondary"
>
  Hello world ;)
</p>
`;

exports[`Typography Should render normal secondary display1 typography 1`] = `
.c0 {
  line-height: 1.6;
  color: #373d3f;
  opacity: .65;
  font-size: 2.5rem;
}

<p
  className="c0"
  type="secondary"
>
  Hello world ;)
</p>
`;

exports[`Typography Should render normal secondary display2 typography 1`] = `
.c0 {
  line-height: 1.6;
  color: #373d3f;
  opacity: .65;
  font-size: 2rem;
}

<p
  className="c0"
  type="secondary"
>
  Hello world ;)
</p>
`;

exports[`Typography Should render normal secondary h2 typography 1`] = `
.c0 {
  line-height: 1.6;
  color: #373d3f;
  opacity: .65;
  font-size: 1.3rem;
  font-weight: bold;
}

<p
  className="c0"
  type="secondary"
>
  Hello world ;)
</p>
`;

exports[`Typography Should render reverse default default typography 1`] = `
.c0 {
  line-height: 1.6;
  color: #fff;
}

<p
  className="c0"
  type="default"
>
  Hello world ;)
</p>
`;

exports[`Typography Should render reverse default display1 typography 1`] = `
.c0 {
  line-height: 1.6;
  color: #fff;
  font-size: 2.5rem;
}

<p
  className="c0"
  type="default"
>
  Hello world ;)
</p>
`;

exports[`Typography Should render reverse default display2 typography 1`] = `
.c0 {
  line-height: 1.6;
  color: #fff;
  font-size: 2rem;
}

<p
  className="c0"
  type="default"
>
  Hello world ;)
</p>
`;

exports[`Typography Should render reverse default h2 typography 1`] = `
.c0 {
  line-height: 1.6;
  color: #fff;
  font-size: 1.3rem;
  font-weight: bold;
}

<p
  className="c0"
  type="default"
>
  Hello world ;)
</p>
`;

exports[`Typography Should render reverse secondary default typography 1`] = `
.c0 {
  line-height: 1.6;
  color: #fff;
  opacity: .65;
}

<p
  className="c0"
  type="secondary"
>
  Hello world ;)
</p>
`;

exports[`Typography Should render reverse secondary display1 typography 1`] = `
.c0 {
  line-height: 1.6;
  color: #fff;
  opacity: .65;
  font-size: 2.5rem;
}

<p
  className="c0"
  type="secondary"
>
  Hello world ;)
</p>
`;

exports[`Typography Should render reverse secondary display2 typography 1`] = `
.c0 {
  line-height: 1.6;
  color: #fff;
  opacity: .65;
  font-size: 2rem;
}

<p
  className="c0"
  type="secondary"
>
  Hello world ;)
</p>
`;

exports[`Typography Should render reverse secondary h2 typography 1`] = `
.c0 {
  line-height: 1.6;
  color: #fff;
  opacity: .65;
  font-size: 1.3rem;
  font-weight: bold;
}

<p
  className="c0"
  type="secondary"
>
  Hello world ;)
</p>
`;

exports[`Typography Should render secondary typography 1`] = `
.c0 {
  line-height: 1.6;
  color: #373d3f;
  opacity: .65;
}

<p
  className="c0"
  type="secondary"
>
  Hello world :D
</p>
`;
